ldaptor/protocols/pureldap.py

```diff
@@ -33,6 +33,8 @@ def alloc_ldap_message_id():
 
 
 def escape(s):
+    if isinstance(s, (bytes, bytearray)):
+        s = s.decode()
```

Comment:
> In addition, arbitrary binary data may be represented by using the escape sequence syntax by encoding each byte of binary data with the backslash () followed by two hexadecimal digits. For example, the four-byte value 0x00000004 is encoded as \00\00\00\04 in a filter string.


see the example in https://www.ietf.org/rfc/rfc2254.html#section-5

Reply:
thx for the hint. so we need to fix this:

```
s = s.replace("\\", r"\5c")
builtins.TypeError: a bytes-like object is required, not 'str'
```